include/arch/x86/multiboot.h

```diff
@@ -36,7 +36,18 @@ struct multiboot_info {
 
 extern struct multiboot_info multiboot_info;
 
-extern void z_multiboot_init(struct multiboot_info *);
+#ifdef CONFIG_MULTIBOOT_INFO
+
+void z_multiboot_init(struct multiboot_info *info);
```

Comment:
Since this is doing multiple duty now as both "multiboot" and "our own boot hook", maybe we should rename it?

Reply:
Or I could test the arg pointer that is supposed to hold multiboot_info or efi_sys_table: if NULL then it's neither multiboot nor efi.

Reply:
Perhaps this is a bit confusing? I don't understand why we have this #ifdef CONFIG_MULTIBOOT_INFO here, it seems to be expected - line 37 defines the struct multiboot_info, which to me indicates we have it configured. I'm sure I'm missing something, sorry!

Reply:
It's just to avoid using #ifdef extensively in the .c files, I prefer using if (IS_ENABLED()) { ... }, so in order to do it I need to enclose the exposed functions with #ifdef in the .h files. (the structures, and macros don't need to)